### PR TITLE
e2e: trim ginkgo.By entries from `system-out` JUnit

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

ginkgo.By has two purposes:
- log a message to the normal test output
- record a ReportEntry that shows when the step started

The second part is not very useful and becomes a nuisance in Spyglass because
it offers an additional "open stdout" button which doesn't really help with
debugging a problem. By filtering out those entries for successful and
unsuccessful tests (and only those entries!), `system-out` never gets added in
most cases.

If a test adds other report entries, for example measurements, they become
visible in Spyglass under "open stdout". Currently they are not shown.

#### Special notes for your reviewer:

This used to be part of https://github.com/kubernetes/kubernetes/pull/111627.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @chendave @aojea

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -201,14 +201,25 @@ func writeJUnitReport(report ginkgo.Report) {
 		// the test output captured in build-log.txt has all of this
 		// information, so we don't need it in the XML.
 		if specReport.State != types.SpecStateFailed {
-			// strip the "system-error" if the testcase is not failed.
```
Comment (pohly) [inline test/e2e/e2e_test.go:204]: These comments seem redundant to me. I can also leave them in. :shrug:

**Comment by aojea** [comment]
this has to wait for the revert, right?

https://github.com/kubernetes/kubernetes/pull/111668

**Comment by pohly** [comment]
Yes. I need to rebase it on top of the new https://github.com/kubernetes/kubernetes/pull/111669.

**Comment by pohly** [comment]
I copied these changes directly into https://github.com/kubernetes/kubernetes/pull/111669.

/close

**Pull request closed without merging**